pkg/controller/pxcrestore/restore_test.go

```diff
@@ -0,0 +1,242 @@
+			bcp:         s3Bcp,
+			expectedErr: "failed to validate backup existence: failed to list objects: failListObjects",
+			objects: []runtime.Object{
+				crSecret,
+				s3Secret,
+			},
+			fakeStorageClientFunc: func(opts storage.Options) (storage.Storage, error) {
+				return &fakeStorageClient{failListObjects: true}, nil
+			},
+		},
+		{
+			name:        "s3 with empty bucket",
+			bcp:         s3Bcp,
+			expectedErr: "failed to validate backup existence: backup not found",
+			objects: []runtime.Object{
+				crSecret,
+				s3Secret,
+			},
+			fakeStorageClientFunc: func(opts storage.Options) (storage.Storage, error) {
+				return &fakeStorageClient{emptyListObjects: true}, nil
+			},
+		},
+		{
+			name: "s3 pitr",
+			bcp:  s3Bcp,
+			cr: updateResource(cr, func(cr *api.PerconaXtraDBClusterRestore) {
+				cr.Spec.PITR = &api.PITR{
+					BackupSource: &api.PXCBackupStatus{
+						StorageName: s3Bcp.Spec.StorageName,
+					},
+				}
+			}),
+			cluster: updateResource(cluster, func(cluster *api.PerconaXtraDBCluster) {
+				cluster.Spec.Backup.PITR = api.PITRSpec{
+					Enabled:     true,
+					StorageName: s3Bcp.Spec.StorageName,
+				}
+			}),
+			objects: []runtime.Object{
+				crSecret,
+				s3Secret,
+			},
+		},
+		{
+			name: "azure",
+			bcp:  azureBcp,
+			objects: []runtime.Object{
+				crSecret,
+				azureSecret,
+			},
+		},
+		{
+			name:        "azure without secrets",
+			bcp:         azureBcp,
+			expectedErr: "failed to validate job: secrets azure-secret, test-cluster-secrets not found",
+		},
+		{
+			name: "azure pitr",
+			bcp:  azureBcp,
+			cr: updateResource(cr, func(cr *api.PerconaXtraDBClusterRestore) {
+				cr.Spec.PITR = &api.PITR{
+					BackupSource: &api.PXCBackupStatus{
+						StorageName: azureBcp.Spec.StorageName,
+					},
+				}
+			}),
+			cluster: updateResource(cluster, func(cluster *api.PerconaXtraDBCluster) {
+				cluster.Spec.Backup.PITR = api.PITRSpec{
+					Enabled:     true,
+					StorageName: azureBcp.Spec.StorageName,
+				}
+			}),
+			objects: []runtime.Object{
+				crSecret,
+				azureSecret,
+			},
+		},
+		{
+			name:        "azure with failing storage client",
+			bcp:         azureBcp,
+			expectedErr: "failed to validate backup existence: list blobs: failListObjects",
+			objects: []runtime.Object{
+				crSecret,
+				azureSecret,
+			},
+			fakeStorageClientFunc: func(opts storage.Options) (storage.Storage, error) {
+				return &fakeStorageClient{failListObjects: true}, nil
+			},
+		},
+		{
+			name:        "azure with empty bucket",
+			bcp:         azureBcp,
+			expectedErr: "failed to validate backup existence: no backups found",
+			objects: []runtime.Object{
+				crSecret,
+				azureSecret,
+			},
+			fakeStorageClientFunc: func(opts storage.Options) (storage.Storage, error) {
+				return &fakeStorageClient{emptyListObjects: true}, nil
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if tt.cr == nil {
```

Comment:
Why don't you set this in the tests themselves rather than updating them like this? The same goes for `tt.cluster`.

Reply:
165d99a